SplitFiles/bread.c

```diff
@@ -31,12 +32,176 @@ BreadType*  rand_bread_type(BreadBin* bin) {
             }
         }
     }
+    //This should never be reached
+    printf("Error in rand_bread_type");
+    return rand_bread_type(bin);
+}
+
+BreadType* get_bread_type(BreadBin* bin, unsigned char typeLetter) {
+    if (typeLetter < 'A' || typeLetter > 'Z') {
+        return NULL;
+    }
+    int index = typeLetter - (unsigned char)'A';
+    return bin->breadTypes[index];
+}
+
+BreadState* get_bread(BreadBin* bin, unsigned char typeLetter) {
+    //int index = typeLetter - (unsigned char)'A';
+    //BreadType* type = *(bin->breadTypes + index);
+    BreadType* type = get_bread_type(bin, typeLetter);
```

Comment:
the bread _types_ are static, but the BreadStates are dynamic based on the BreadType.  That said, this method could have a pointer to a BreadType passed to it instead of the letter, to save this get_bread_type call.  The letter-as-arg decision was about working with the previous implementation.